File: backend/internal/domain/company_application.go
package domain

import "time"

// CompanyApplication は未登録の企業担当者がログイン前の公開フォームから出す「利用申請」。
// super_admin が一覧で確認し、問題なければ既存の招待フローで company_admin を招待する。
type CompanyApplication struct {
	ID            uint64    `json:"id"`
	CompanyName   string    `json:"companyName"`
	ApplicantName string    `json:"applicantName"`
	Email         string    `json:"email"`
	Message       string    `json:"message"`
	Status        string    `json:"status"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

const (
	CompanyApplicationStatusPending  = "pending"
	CompanyApplicationStatusApproved = "approved"
	CompanyApplicationStatusRejected = "rejected"
)

// NotificationTypeCompanyApplication は企業申請が届いたことを super_admin に知らせる通知の Type。
const NotificationTypeCompanyApplication = "company_application"

File: backend-java/src/main/java/com/normanblog/frestyle/entity/CompanyApplication.java
package com.normanblog.frestyle.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 未登録の企業担当者がログイン前に出す「利用申請」。
 *
 * <p>super_admin が一覧で確認し、問題なければ招待フローで company_admin を招待する入口になる。
 */
@Entity
@Table(name = "company_applications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CompanyApplication {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "company_name", nullable = false)
  private String companyName;

  @Column(name = "applicant_name", nullable = false)
  private String applicantName;

  @Column(nullable = false)
  private String email;

  @Column(columnDefinition = "text")
  private String message;

  @Column(nullable = false)
  private String status;

  @Column(name = "created_at")
  private Instant createdAt;

  @Column(name = "updated_at")
  private Instant updatedAt;
}
